Add status and method in calendar event

giuseppenucifora opened the pull request:
- Add status property in Event
- Add method property in Event

markuspoerschke commented:
Hey Giuseppe,

thanks for your contribution. There are some problem that needs to be solved before I can merge this pull request:

- [ ] The CI-Pipeline is failing. Please [check the logs](https://github.com/markuspoerschke/iCal/pull/300/checks?check_run_id=2996804491) and resolve the issues. (You can fix the code style running `make fix` and executing the tests locally using `make test`.)
- [ ] Please update the documentation as well.
- [x] A new entry to the changelog is needed to inform other developers about this new feature.

markuspoerschke commented on src/Domain/Entity/Event.php:
```diff
@@ -21,6 +21,17 @@
 
 class Event
 {
+    const EVENT_STATUS_CONFIRMED = 'CONFIRMED';
```
Instead of using constants, please use enums as shown here: https://github.com/markuspoerschke/iCal/blob/2.x/src/Domain/Enum/TimeZoneTransitionType.php

markuspoerschke commented on src/Presentation/Factory/EventFactory.php:
```diff
@@ -73,8 +73,10 @@ public function createComponent(Event $event): Component
-        yield new Property('UID', new TextValue((string) $event->getUniqueIdentifier()));
+        yield new Property('UID', new TextValue((string)$event->getUniqueIdentifier()));
         yield new Property('DTSTAMP', new DateTimeValue($event->getTouchedAt()));
+        yield new Property('STATUS', new TextValue($event->getStatus()));
```
Since `STATUS` is optional, please add a "if condition" that will check if the property needs to be added:

```suggestion
        if ($event->hasStatus()) {
            yield new Property('STATUS', new TextValue($event->getStatus()));
        }
```

The same applies to `METHOD`.

giuseppenucifora replied:
Sure, now i fix all issues, thank you very much.

giuseppenucifora replied:
Fixed as you suggest

markuspoerschke commented on src/Presentation/Factory/EventFactory.php:
```diff
@@ -111,7 +113,7 @@ protected function getProperties(Event $event): Generator
     protected function getComponents(Event $event): Generator
     {
         yield from array_map(
-            fn (Alarm $alarm) => $this->alarmFactory->createComponent($alarm),
```
The code style was changed here. That is one of the reasons, why the CI pipeline fails.

You can fix the code style by running `make fix` locally.

giuseppenucifora commented:
@markuspoerschke as you rightly suggested, I refactor the code by adding two enum for method and status but now launching test , the phpmd test gave this error :

`phpmd ./src text rulesets.xml
/Volumes/External/Progetti/GiuseppeNucifora/GIT/PHP/iCal/src/Presentation/Factory/EventFactory.php:74   The method getProperties() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.
/Volumes/External/Progetti/GiuseppeNucifora/GIT/PHP/iCal/src/Presentation/Factory/EventFactory.php:74   The method getProperties() has an NPath complexity of 512. The configured NPath complexity threshold is 200.
make: *** [test-phpmd] Error 2
`

can you help me to solve? Scuse me, but unfortunately i don't know this. :)

giuseppenucifora committed:
- Add status property in Event
- Add method property in Event
- Update Unit test to test new properties

giuseppenucifora committed:
- update changelog

markuspoerschke commented:
> `phpmd ./src text rulesets.xml /Volumes/External/Progetti/GiuseppeNucifora/GIT/PHP/iCal/src/Presentation/Factory/EventFactory.php:74 The method getProperties() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. /Volumes/External/Progetti/GiuseppeNucifora/GIT/PHP/iCal/src/Presentation/Factory/EventFactory.php:74 The method getProperties() has an NPath complexity of 512. The configured NPath complexity threshold is 200. make: *** [test-phpmd] Error 2 `

This is a violated rule from PHPMD. Since the Factory class will assemble a lot of properties, I think we can ignore the error message. You could add a `@SuppressWarnings("NPathComplexity")` to the method. See https://phpmd.org/documentation/suppress-warnings.html

markuspoerschke commented on src/Domain/Enum/MethodType.php:
```diff
@@ -0,0 +1,42 @@
+
+namespace Eluceo\iCal\Domain\Enum;
+
+final class MethodType
```
```suggestion
final class Method
```

The suffix "type" is not needed for this Enum.

giuseppenucifora replied:
Sure, i'm updating.

markuspoerschke commented on src/Domain/Enum/StatusType.php:
```diff
@@ -0,0 +1,48 @@
+
+namespace Eluceo\iCal\Domain\Enum;
+
+final class StatusType
```
```suggestion
final class Status
```

giuseppenucifora replied:
Sure, i'm updating.

markuspoerschke committed:
Cleanup Makefile and avoid composer install if vendor folder is present (#302)

* remove PHP-CS-Cache v3 with make clean
* dont use phony target for composer dependencies:
  don't install composer dependencies if already installed
  this will speedup `make test` locally
* dont install node_modules if present when `make fix-prettier` is called

giuseppenucifora committed:
Change StatusType enum tu Status
Change MethodType enum tu Method
Add @SuppressWarnings("PHPMD.NPathComplexity") in getProperties function of EventFactory class
Add @SuppressWarnings("PHPMD.CyclomaticComplexity") in getProperties function of EventFactory class
